FaunaDB.Client/Types/Encoder.cs

```diff
@@ -85,6 +87,24 @@ Value EncodeIntern(object obj, Type forceType)
         {
             var type = obj.GetType();
 
+            if (forceType == typeof(StringV))
+            {
+                switch (Type.GetTypeCode(type))
+                {
+                    case TypeCode.Object:
+                        if (typeof(byte[]).IsInstanceOfType(obj))
+                            return JsonConvert.SerializeObject(obj);
+
+                        if (typeof(IDictionary).IsInstanceOfType(obj))
+                            return JsonConvert.SerializeObject(obj);
+
+                        if (typeof(IEnumerable).IsInstanceOfType(obj))
+                            return JsonConvert.SerializeObject(obj);
+                        break;
+                }
```

Comment:
please add a test for this specific "branch"

Reply:
removed no longer needs test